infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -109,11 +110,18 @@
     private final AtomicLong counter = new AtomicLong(1);
 
     /**
-     * Number of workers for failure detector. Two workers used by dafult:
+     * Number of workers for failure detector. Three workers used by default:
      * - failure/healing detection
      * - bootstrap sequencer
+     * - merge segments
      */
-    private final int detectionWorkersCount = 2;
+    private final int detectionWorkersCount = 3;
```

Comment:
why do we need an extra thread ?

Reply:
/**
Number of workers for failure detector. Three workers used for the following tasks:
- failure/healing detection
- bootstrap sequencer
- merge segments